### DISCUSSION TITLE
Is there a way to change a default GLOBAL configuration value for a cluster?

### DISCUSSION THREAD

**Discussion by jekramer** [JanusGraph/janusgraph#3886] [Q&A]
We need `log.user.ttl` config property set for our JanusGraph cluster. I know that it is a GLOBAL config, so if we want to change it we need to use the management system. If I do that, it changes the value only for the keyspace I am currently in. When I start a new keyspace in Gremlin console with `log.ttl.user` set in the properties file though, the value doesn't register until I use the management system to set it. Is it possible to set the default for the whole cluster so that new keyspaces will initialize with a value? They all currently start with `log.user.ttl=null`

Any guidance would be greatly appreciated!

**Comment by hadoopmarc** [comment]
I do not think it possible without code changes, because the `null` default value is hard coded. It is possible to subclass JanusGraphFactory and modify the configs in the overrided constructors, but it would be hard to prevent cluster participants from still working with the original factory. If you would be willing to provide a PR for a more rigourous solution, you can open a ticket on github and discuss possible solutions with the project committers.

**Comment by jekramer** [comment]
@hadoopmarc it actually seems like GLOBAL wasn't the problem here, it was the format I was using to set the ttl in the properties file. @jianhuang1648678534 opened issue #3900 describing where he noticed the failure
